web3-functions/index.ts

```diff
@@ -0,0 +1,305 @@
+import {
+  Web3Function,
+  Web3FunctionContext,
+} from "@gelatonetwork/web3-functions-sdk";
+import { BigNumber, Contract } from "ethers";
+import { arrayify, toUtf8String, formatBytes32String } from "ethers/lib/utils";
+import { WrapperBuilder } from "@redstone-finance/evm-connector";
+
+const ORACLE_ABI = [
+  "function updateDataFeedsValuesPartial(bytes32[]) public",
+  "function getLastUpdateDetails(bytes32) public view returns (uint256, uint256, uint256)",
+  "function getLivePrice(bytes32[]) public view returns (uint256[], uint256)",
+];
+
+const redstone = require("redstone-protocol");
+
+type DataFeed = {
+  symbol: string;
+  id: string;
+  livePrice: BigNumber;
+  timestamp: number;
+  storedPrice: BigNumber;
+  storedTimestamp: number;
+};
+
+const isDebugMode = false;
+
+Web3Function.onRun(async (context: Web3FunctionContext) => {
+  const { userArgs, multiChainProvider } = context;
+
+  const provider = multiChainProvider.default();
+
+  const dataServiceId = userArgs.dataServiceId as string;
+  const dataFeedIdsString = userArgs.symbols as string[];
+  const dataFeedIdsBytes32 = dataFeedIdsString.map((id) =>
+    formatBytes32String(id)
+  );
+  const oracleAddress = userArgs.oracleAddress as string;
+
+  const oracle = new Contract(oracleAddress, ORACLE_ABI, provider);
+
+  let dataFeedIds = new Map<string, DataFeed>();
+  for (const id of dataFeedIdsString) {
+    dataFeedIds.set(id, {
+      symbol: id,
+      id: formatBytes32String(id),
+      livePrice: BigNumber.from(0),
+      timestamp: 0,
+      storedPrice: BigNumber.from(0),
+      storedTimestamp: 0,
+    });
+  }
+  conditionalLog(isDebugMode, "Data feed ids: ", dataFeedIds);
+
+  // Wrap contract with redstone data service
+  var wrappedOraclePrimaryProd;
```

Comment:
no `var`, it should be replaced with `const` preferably, and `const` doesn't work, then `let`. 

same for other places

Reply:
Done.